test/lib/jdk/test/lib/hprof/parser/Reader.java

```diff
@@ -164,9 +163,8 @@ public static String getStack(String heapFile, int debugLevel)
                     throw new IOException("Cannot decompress the compressed hprof file", e);
                 }
                 // Check dump data header and print stack trace.
-                try {
-                    PositionDataInputStream in2 = new PositionDataInputStream(
-                        new BufferedInputStream(new FileInputStream(out)));
+                try (PositionDataInputStream in2 = new PositionDataInputStream(
+                        new BufferedInputStream(new FileInputStream(out)))) {
```

Comment:
Similarly here. You first need to define and create the FIS, then the BIS, then the PDIS.